### PR TITLE
[CSL-1501] LRC test fails sometimes

### PR DESCRIPTION
**Pull request by pva701**
I tested with `modifyMaxSuccess (const 100)`, it works

### CODE REVIEW

**Comment by pva701** [comment]
@gromakovsky simplified, take a look plz

File: test/Test/Pos/Lrc/WorkerSpec.hs [hunk 1]
```diff
@@ -75,61 +78,66 @@ allRichmenComponents =
...
-        let poorStake = mkCoin baseN `unsafeMulCoin` poorStakeN
-        let k1 = (r - poorStakeN) `div` 3
-        -- This ensures that no rich stake is less than threshold and
-        -- no more than k1, which is the remained stake (w/o poor)
-        -- divided by three.
-        [richStake1,richStake2] <-
-            replicateM 2 $
-            (`unsafeMulCoin` (baseN :: Int)) . mkCoin <$>
-            choose (poorStakeN, k1)
+        -- Substract 1 to avoid corner case when poorStake == thresholdN
```
Comment (gromakovsky) [inline test/Test/Pos/Lrc/WorkerSpec.hs]: This comment is obsolete.

File: test/Test/Pos/Lrc/WorkerSpec.hs [hunk 2]
```diff
@@ -75,61 +78,66 @@ allRichmenComponents =
...
+    identityDistr _ _                       = error "Unexpected address type"
+
+    genesisContextSimple :: [AddrDistribution] -> GenesisContext
+    genesisContextSimple addrDistr = do
+        let balances = concatAddrDistrs addrDistr
+        let utxoEntry (addr, coin) =
+                ( TxIn (unsafeHash addr) 0
+                , TxOutAux (TxOut addr coin) (identityDistr addr coin))
+        GenesisContext (GenesisUtxo $ M.fromList $ map utxoEntry balances)
+                       (GenesisWStakeholders mempty)
```
Comment (gromakovsky) [inline test/Test/Pos/Lrc/WorkerSpec.hs]: It would be good to comment why we don't care about genesis stakeholders.

Comment (pva701) [inline test/Test/Pos/Lrc/WorkerSpec.hs]: tbh I don't the reason for sure

**Comment by gromakovsky** [comment]
Merged, etc. Just need CI to pass.

**Comment by pva701** [comment]
sps gromak <3
